### ISSUE TITLE
Paginator for S3 `ListParts` does not terminate

### ISSUE THREAD

**Issue by jbelkins** [awslabs/aws-sdk-swift#1021]
### Describe the bug

This paginated operation requires customization to work properly, i.e.:
https://github.com/aws/aws-sdk-go-v2/pull/1141

### Expected Behavior

Paginator for S3 `ListParts` terminates when all items have been listed.

### Current Behavior

S3 `ListPartsPaginated` loads new pages in an infinite loop as `isTruncated` is not implemented as a valid paginator.

### Reproduction Steps

Perform paginated operation on AWS SDK for Swift.  Observe paginator behavior.

```
let bucket = "dayaffe-testing"
let objectKey = "list-parts-test"
let client = try S3Client(region: "us-west-2")
let chunk = Data(repeating: 33, count: 5 * 1024 * 1024) // Parts must be at least 5MB
let expectedParts = Array(1...10)

let output = try await client.createMultipartUpload(input: .init(bucket: bucket, key: objectKey))
var uploadedParts: [S3ClientTypes.CompletedPart] = []
for idx in expectedParts {
    let part = try await client.uploadPart(input: .init(
        body: ClientRuntime.ByteStream.from(data: chunk),
        bucket: bucket,
        key: objectKey,
        partNumber: idx,
        uploadId: output.uploadId
    ))
    let completedPart = S3ClientTypes.CompletedPart(eTag: part.eTag, partNumber: idx)
    uploadedParts.append(completedPart)
}
print("LISTING PARTS")
let listedParts = client.listPartsPaginated(input: .init(
    bucket: bucket,
    key: objectKey,
    maxParts: 5,
    uploadId: output.uploadId
))
await print(try listedParts.parts()) // this will infinite loop
print("LISTED")
// Complete the multipart upload
let completedUpload = S3ClientTypes.CompletedMultipartUpload(parts: uploadedParts)
let completeMultipartUploadInput = CompleteMultipartUploadInput(
    bucket: bucket,
    key: objectKey,
    multipartUpload: completedUpload,
    uploadId: output.uploadId
)
_ = try await client.completeMultipartUpload(input: completeMultipartUploadInput)
```

### Possible Solution

Implement a customization for this paginator that supports `isTruncated`

### Additional Information/Context

SWIFT-153

### AWS SWIFT SDK version used

latest

### Compiler and Version used

latest

### Operating System and version

latest

**Comment by dayaffe** [comment]
Note: This may be reversed in future as Smithy team is currently in talks with S3 to get them to stop using `isTruncated`

**Comment by jbelkins** [comment]
Released in AWS SDK for Swift 0.32.0
